src/pudl/io_managers.py

```diff
@@ -264,28 +260,17 @@ def _handle_pandas_output(self, context: OutputContext, df: pd.DataFrame):
         sa_table = self._get_sqlalchemy_table(table_name)
         engine = self.engine
 
-        # TODO (bendnorman): I included this if else statement for the analysis table example.
-        if sa_table is None:
-            with engine.connect() as con:
-                # Remove old table records before loading to db
-                df.to_sql(
-                    table_name,
-                    con,
-                    if_exists="replace",
-                    index=False,
-                )
-        else:
-            with engine.connect() as con:
-                # Remove old table records before loading to db
-                con.execute(sa_table.delete())
-
-                df.to_sql(
-                    table_name,
-                    con,
-                    if_exists="append",
-                    index=False,
-                    dtype={c.name: c.type for c in sa_table.columns},
-                )
+        with engine.connect() as con:
+            # Remove old table records before loading to db
+            con.execute(sa_table.delete())
+
+            df.to_sql(
+                table_name,
+                con,
+                if_exists="append",
```

Comment:
If we just deleted the table, will it ever exist to append to here? Or does delete just remove the data in the table and not the table and its definition?

Reply:
The [`delete()` function](https://docs.sqlalchemy.org/en/20/tutorial/data_update.html#the-delete-sql-expression-construct) just deletes the data, it does not drop the table. This way the table schemas are preserved across runs.